lib/util.js

```diff
@@ -1627,6 +1644,32 @@ util.decode64 = function(input) {
   return output;
 };
 
+/**
+ * Decodes input according to the "Base64url Encoding" format as specified
+ * in JSON Web Signature (JWS) RFC7517. A URL safe character set is used and
+ * trailing '=', line breaks, whitespace, and other characters are omitted.
+ *
+ * @param input the data to decode.
+ *
+ * @return the decoded value.
+ */
+util.decodeBase64Url = function decodeBase64Url(input) {
+  let normalInput = input
+    .replace(/-/g, '+')
+    .replace(/_/g, '/');
+  const mod4 = normalInput.length % 4;
+  if(mod4 === 0) {
```

Comment:
it appears that `mod4 === 1` is the error condition, maybe we can do without the === 0 check and be explicit about the error condition.

Reply:
Sure!